truffle-config.js

```diff
@@ -14,14 +14,14 @@ module.exports = {
 
   compilers: {
     solc: {
-      // version: "0.5.1",    // Fetch exact version from solc-bin (default: truffle's version)
+      version: "0.4.24",    // Fetch exact version from solc-bin (default: truffle's version)
       // docker: true,        // Use "0.5.1" you've installed locally with docker (default: false)
       // settings: {          // See the solidity docs for advice about optimization and evmVersion
       //  optimizer: {
       //    enabled: false,
       //    runs: 200
       //  },
-      //  evmVersion: "byzantium"
+      evmVersion: "constantinople"
```

Comment:
we need to build against `constantinople`
otherwise
```Warning: The "shl" instruction is only available for Constantinople-compatible VMs. You are currently compiling for "byzantium", where it will be interpreted as an invalid instruction.
            output := gt(and(sload(playerInput_slot), shl(i, 1)), 0)```

Reply:
We do not need ganache to target `constantinople`.